Move SemVerHandler to analysis to parsing to increase modularity...

rocky opened the pull request:
If ever you want to separate building analysis from scanning/parsing/ast, this will help.

<!--### Your checklist for this pull request

Please review the [guidelines for contributing](http://solidity.readthedocs.io/en/latest/contributing.html) to this repository.

Please also note that this project is released with a [Contributor Code of Conduct](CONDUCT.md). By participating in this project you agree to abide by its terms.
-->

### Description

<!--
Please explain the changes you made here.

Thank you for your help!
-->

### Checklist
- [x] Code compiles correctly
- [x] All tests are passing
- [ ] New tests have been created which fail without the change (if possible)
- [ ] README / documentation was extended, if necessary
- [ ] Changelog entry (if change is visible to the user)
- [x] Used meaningful commit messages

chriseth reviewed (changes requested):
Since the SyntaxChecker also relies on the SemVerHandler, I think it would be best to move it all the way down to liblangutil. SemVerChecker.h includes `parsing/Token.h` but that is actually an "alias" for `liblangutil/Token.h`.

rocky commented:
Ah - okay. Thanks for the information. Will make the changes later today.

stackenbotten commented:
There was an error when running `test_check_style` for commit `389a56be693d324fdeddb2ec36ad12cb93e0b94a`:
```
Coding style error:
 liblangutil/SemVerHandler.h:36: class SemVerError: dev::Exception

```
Please check that your changes are working as intended.

chriseth commented:
Please squash into one commit.

rocky committed:
Move SemVerHandler to analysis to parsing to increase modularity...

If ever you want to separate building analysis from scanning/parsing/ast, this will help.

stackenbotten commented:
There was an error when running `test_check_style` for commit `ebe6413f39f62e3a43711ff29461fe0411b4a8b1`:
```
Error: Trailing whitespace found:
 liblangutil/ParserBase.cpp.rej:6: 

```
Please check that your changes are working as intended.

rocky commented:
@chriseth ok I think everything is squashed to one commit now.

chriseth commented on liblangutil/SemVerHandler.h:
```diff
@@ -31,6 +31,8 @@ namespace dev
 namespace solidity
```
The namespace should change now, too.

rocky replied:
should be hanled in e0de5aa - but please double check.

chriseth commented on liblangutil/SemVerHandler.h:
```diff
@@ -22,15 +22,15 @@
+namespace langutil
 {
 
+using ::langutil::Token;
```
This should have no effect now and should be deleted.

rocky replied:
Ok - how about now in b42446 ?

rocky commented:
@chriseth BTW after this, the next up on my list of Modularity things is moving `Version.h` from `libsolsolidity/interface` to some place eariler, like `libsolidity/parsing`, or if you can suggest some other place, that would be good.
